sound/soc/sof/ipc4-control.c

```diff
@@ -35,23 +37,34 @@ static int sof_ipc4_set_get_kcontrol_data(struct snd_sof_control *scontrol, bool
 		return -ENOENT;
 	}
 
+	if (lock)
+		mutex_lock(&swidget->setup_mutex);
+	else
```

Comment:
isnt the else redundant here?

Reply:
well one could argue that if you test that the mutex_lock did work it's also redundant...